# 1. Using the data provided in Bronchitis.csv generate a Linear Model, evaluate the model, and provide the evidence of the appropriateness of the model. The aim of the study was to define if the probability of having Bronchitis is influenced by smoking and/or pollution. The data are stored contains information on 212 participants.

In [3]:
library(ggplot2)

loading library

In [5]:
data <- read.csv('Bronchitis.csv')

read csv file and then print out sample of data to see if its read correctly

In [6]:
head(data)

bron cigs poll
1 0    5.15 67.1
2 1    0.00 66.9
3 0    2.50 66.7
4 0    1.75 65.8
5 0    6.75 64.4
6 0    0.00 64.4

In [18]:
total_counts_of_bron <- table(data$bron)

In [21]:
cat("Total count of non-bronchitis cases", total_counts_of_bron[1], "\n")
cat("Total count of bronchitis cases:", total_counts_of_bron[2], "\n")

Total count of non-bronchitis cases 166 
Total count of bronchitis cases: 46 


I just wanted to see the total count of cases for bronchitis so I can just better understand the data that I was given. 

In [23]:
model <- glm(bron ~ cigs + poll, data = data, family = binomial)

Using GLM function which is generalized linear model with family as binomial since output is true or false on bronchitis. Cigs and Polls are the explanatory variables

In [25]:
summary(model)


Call:
glm(formula = bron ~ cigs + poll, family = binomial, data = data)

Coefficients:
             Estimate Std. Error z value Pr(>|z|)    
(Intercept) -10.08491    2.95100  -3.417 0.000632 ***
cigs          0.21169    0.03813   5.552 2.83e-08 ***
poll          0.13176    0.04895   2.692 0.007113 ** 
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 221.78  on 211  degrees of freedom
Residual deviance: 174.21  on 209  degrees of freedom
AIC: 180.21

Number of Fisher Scoring iterations: 5


# Summary
Cigs and Pollution are highly significant since p value is either (p < 0.001 or p < .01). 

Cigs and poll have estimate that shows that every cig smokes per day, the log-odds of having bronchitis increases by .21169. Pollution is .1376. 

Residual deviance being lower than null deviance shows that cig and polls fit the data between than the model just being null. 

Based on everything Cigs and Poll are significant predictors of bronchitis

I am not doing a fit reduced model since both my predictor variables are significant. 

In [33]:
predicted_probabilities_bron <- predict(model, type = "response")

This will be predicted true or false for bronchitis. Its already trained in the output of true and false for bronchitis so this new line of code is going to make predictions on the data again and attempt to get the bron now without seeing it in the data.

In [34]:
threshold <- 0.5
predicted_classes <- ifelse(predicted_probabilities_bron > threshold, 1, 0)


Every row is checked and giving a probability percentage and if its above or equal to 50 percent its true and if below its false. Predicted classes grabs from data already established and creates into another data.  

In [55]:
conf_matrix <- table(Predicted = predicted_classes, Actual = data$bron)
print(conf_matrix)


         Actual
Predicted   0   1
        0 156  32
        1  10  14


This will check my prediction dataset with actual dataset.  Checking true and false on both for comparison. Displayed is true negatives, true negatives, false positive and false negative. Output here shows 156 true negatives and 14 true positives. 170 correct predictions and 42 wrong predictions. 

In [57]:
accuracy <- sum(diag(conf_matrix)) / sum(conf_matrix)
print(paste("Accuracy percentage:", accuracy))

[1] "Accuracy percentage: 0.80188679245283"


170 / 210 = .80188  - 80 percent is a good percentage, however, more than half of true statements are false postives which dont seem good to me. 

In [59]:
install.packages("caret")
install.packages("lattice")


The downloaded binary packages are in
	/var/folders/rd/jfxc79756t9g0g5f4ytyb6l40000gn/T//RtmpAjvAUa/downloaded_packages

The downloaded binary packages are in
	/var/folders/rd/jfxc79756t9g0g5f4ytyb6l40000gn/T//RtmpAjvAUa/downloaded_packages


In [60]:
library(caret)
library(lattice)


In [65]:
confusion_matrix <- confusionMatrix(as.factor(predicted_classes), as.factor(data$bron))
print(confusion_matrix)

Confusion Matrix and Statistics

          Reference
Prediction   0   1
         0 156  32
         1  10  14
                                          
               Accuracy : 0.8019          
                 95% CI : (0.7418, 0.8533)
    No Information Rate : 0.783           
    P-Value [Acc > NIR] : 0.283466        
                                          
                  Kappa : 0.2951          
                                          
 Mcnemar's Test P-Value : 0.001194        
                                          
            Sensitivity : 0.9398          
            Specificity : 0.3043          
         Pos Pred Value : 0.8298          
         Neg Pred Value : 0.5833          
             Prevalence : 0.7830          
         Detection Rate : 0.7358          
   Detection Prevalence : 0.8868          
      Balanced Accuracy : 0.6221          
                                          
       'Positive' Class : 0               
                              

95% Confidence rate is between 74 percent to 85 percent. My model is 80.19% and No information Rate is 78 percent. Which means model is slighty higher than NIR. NIR is going to predict only non bron cases. Model is slighty better 

Mcnemar's Test P-Value : 0.001194  False positive is 10 and false negative is 32 which means that their is a high imbalance between between false positives and false negatives. 

P - Value = .2834. Since this is higher than .05. I reject that my models accuracy is better than NIR.  

 Kappa : 0.2951 = this is fair aggreement between models predictions are the actual claffications. Which means that our model is slightly better than chance.
 
 
'Positive' Class : 0 = No Bron

Sensitivity being .9398 means my model is very good at measuring the proportion of no bronchities cases which were correctly identified as not having bronchitis. 

Specificity being .3043 means my model is not good at identifying patients that do have bronchitis. 

positive predictions of no bron cases is .8298 and negative prediction is .5833. Appears my model is better at predicted no bron and is barely over 50 percent at predicting cases where their is bronchitis. This shows that my model is good at predicting no bron and not very good at bron. 

Prevalence : 0.7830          
Detection Rate : 0.7358          
Detection Prevalence : 0.8868          
Balanced Accuracy : 0.6221   

